odl/solvers/advanced/proximal_operators.py

```diff
@@ -143,108 +236,227 @@ def make_prox(tau):
+    """Function to create the proximal operator of G(x) = ind(x >= 0).
+
+    Function for the proximal operator of the functional G(x)=ind(x >= 0) to be
+    initialized.
 
+    If P is the set of non-negative elements, the indicator function of
+    which is defined as
+
+        ind(x >= 0) = {0 if x in P, infinity if x is not in P}
+
+    with x being an element in ``space``.
+
+    The proximal operator of ``tau * F^*`` where ``tau`` is a step size
+    is the point-wise non-negativity thresholding of x
+
+         prox[tau * G](x) = {x if x >= 0, 0 if < 0}
+
+    It is independent of tau and invariant under a positive rescaling of G
+    which leaves the indicator function as it stands.
+
+    Parameters
+    ----------
+    space : `LinearSpace`
+        Domain of the functional G(x)
 
-def proximal_convexconjugate_l2(space, lam=1, g=None):
+    Returns
+    -------
+    prox_factory : `callable`
+        Factory for the proximal operator to be initialized
+
+    See Also
+    --------
+    proximal_clamp
+    """
+
+    return proximal_box_constraint(space, lower=0)
+
+
+def proximal_cconj_l2(space, lam=1, g=None):
     """Proximal operator factory of the convex conjugate of the l2-norm.
 
     Function for the proximal operator of the convex conjugate of the
     functional F where F is the l2-norm
 
-        F(x) =  lam 1/2 ||x - g||_2^2
+        F(x) =  lam ||x - g||_2
 
     with x and g elements in ``space``, scaling factor lam, and given data g.
 
-    The convex conjugate, F_cc, of F is given by
+    The convex conjugate, F^*, of F is given by
 
-        F_cc(y) = 1/lam (1/2 ||y/lam||_2^2 + <y/lam,g>)
+        F^*(y) = {0 if ||x-g|| < lam, infty else}
 
-    The proximal operator of F_cc is given by
+    The proximal operator of ``sigma * F^*`` where ``sigma`` is a step size
+    is given by
 
-        prox_sigma[F_cc](y) = (y - sigma g) / (1 + sigma/lam)
+        prox[sigma * F^*](y) = (x - g) / ||x - g||
 
     Parameters
     ----------
-    space : `DiscreteLp` or `ProductSpace` of `DiscreteLp`
-        Domain of F(x)
-    g : `DiscreteLpVector`
+    space : `LinearSpace`
+        Domain of F(x). Needs to be a hilbert space (have an inner product)
+    g : ``space`` element
         An element in ``space``
     lam : positive `float`
         Scaling factor or regularization parameter
 
     Returns
     -------
-    prox : `Operator`
-        Returns the proximal operator to be initialized
+    prox_factory : `callable`
+        Factory for the proximal operator to be initialized
+
+    Notes
+    -----
+    Most problems are forumlated for the squared norm, in that case use the
+    `proximal_cconj_l2_squared` instead.
+
+    See Also
+    --------
+    proximal_l2 : proximal without convex conjugate
+    proximal_cconj_l2_squared : proximal for squared norm
     """
+
+    prox_l2 = proximal_l2(space, lam=lam, g=g)
+    return proximal_cconj(prox_l2)
+
+
+def proximal_l2(space, lam=1, g=None):
```

Comment:
You are technically correct, but doing this would make the name for some of these even worse than they are already. Further there is no mention of norm in the function name, so this can be interpreted to mean the l2 distance, which would be correct.

I guess we can update the formulations in the documentation where we use the word norm however.